apps/roam/src/components/settings/NodeConfig.tsx

```diff
@@ -40,6 +120,68 @@ const NodeConfig = ({
   });
 
   const [selectedTabId, setSelectedTabId] = useState<TabId>("general");
+  const [tagError, setTagError] = useState("");
+  const [formatError, setFormatError] = useState("");
+  const isValid = !tagError && !formatError;
+
+  const {
+    value: tagValue,
+    handleChange: handleTagChange,
+    handleBlur: handleTagBlurFromHook,
+  } = useDebouncedRoamUpdater(tagUid, node.tag, isValid);
+  const {
+    value: formatValue,
+    handleChange: handleFormatChange,
+    handleBlur: handleFormatBlurFromHook,
+  } = useDebouncedRoamUpdater(formatUid, node.format, isValid);
+
+  const getCleanTagText = (tag: string): string => {
+    return tag.replace(/^#+/, "").trim().toUpperCase();
+  };
+
+  const validate = useCallback((tag: string, format: string) => {
+    const cleanTag = getCleanTagText(tag);
+
+    if (!cleanTag) {
+      setTagError("");
+      setFormatError("");
+      return;
+    }
+
+    const formatWithoutPlaceholders = format.replace(/{[^}]+}/g, "");
+    const formatUpper = formatWithoutPlaceholders.toUpperCase();
+    const formatParts = formatUpper.split(/[^A-Z0-9]/);
+    const hasConflict = formatParts.includes(cleanTag);
```

Comment:
This is checking if the tag appears anywhere as word in the format.  Which means even this would be invalid:

tag: claim
format: [[CLM]] - my claim is: {content}

What we need to do is check if the tag is **referenced** anywhere in the format.  Eg: if it appears in like `[[claim]]` or `#[[claim]]` or (rare) `#[[claim]]`.

I don't think `roamjs-components` has a regex for this, but there is `extractTag` for going the other way that you can use in the function.